### PR TITLE
[Reference cleanup]LPS-162273

### PR DESCRIPTION
**Pull request by JanisZhang**
Hi @dantewang @tinatian,

resend, fixed unit test.

This PR is to convert method references to field references in dynamic-data-mapping-service | Part 2.
https://issues.liferay.com/browse/LPS-162273

previous PR : https://github.com/liferay-core-infra/liferay-portal/pull/1556

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMXMLImpl.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public Fields getFields(
 			List<String> fieldNames)
 		throws PortalException {
 
-		Document document = null;
+		Document document;
```
Comment (dantewang) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMXMLImpl.java:76]: @JanisZhang Was this change required by Source Formatter?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMDataProviderInstanceServiceImpl.java [hunk 2]
```diff
@@ -240,11 +240,6 @@ public DDMDataProviderInstance updateDataProviderInstance(
 			ddmFormValues, serviceContext);
 	}
 
-	@Reference(unbind = "-")
-	protected void setJSONFactory(JSONFactory jsonFactory) {
```
Comment (dantewang) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMDataProviderInstanceServiceImpl.java:244]: @JanisZhang For this kind of removal, please write the commit message like "Remove redundant method reference in xxx as the field is already referenced".

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMXMLImpl.java (hunk 1 above)
Comment (JanisZhang) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMXMLImpl.java:76]: @dantewang Yes

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMDataProviderInstanceServiceImpl.java (hunk 2 above)
Comment (JanisZhang) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMDataProviderInstanceServiceImpl.java:244]: @dantewang Got it,thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMXMLImpl.java (hunk 1 above)
Comment (JanisZhang) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMXMLImpl.java:76]: @dantewang No, It's recommended by IDE.

Comment (dantewang) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMXMLImpl.java:76]: @JanisZhang Please ignore the IDE recommendation. Actually it's a convention in Liferay to give a variable an initial value, even if it's not used.

Comment (JanisZhang) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMXMLImpl.java:76]: @dantewang Ok

**Comment by JanisZhang** [comment]
Hi @dantewang, force-pushed. 

Hi @catarineaguiar,
I drop one SF commit and rewrite 1 commit message. It shouldn't affect the ci results. Thanks

**Review by dantewang — APPROVED** [review]

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
Hi @catarineaguiar ,Could you please help to scan the test results? Thanks

**Comment by catarineaguiar** [comment]
@JanisZhang no regressions found.

**Comment by tinatian** [comment]
FYI @shuyangzhou , see https://github.com/brianchandotcom/liferay-portal/pull/122727
